vsock/src/main.rs

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+mod rxops;
+mod rxqueue;
+mod thread_backend;
+mod txbuf;
+mod vhu_vsock;
+mod vhu_vsock_thread;
+mod vsock_conn;
+
+use clap::Parser;
+use log::{info, warn};
+use std::{
+    convert::TryFrom,
+    sync::{Arc, RwLock},
+};
+use vhost::{vhost_user, vhost_user::Listener};
+use vhost_user_backend::VhostUserDaemon;
+use vhu_vsock::{VhostUserVsockBackend, VsockArgs, VsockConfig};
+use vm_memory::{GuestMemoryAtomic, GuestMemoryMmap};
+
+/// This is the public API through which an external program starts the
+/// vhost-user-vsock backend server.
+pub(crate) fn start_backend_server(vsock_config: VsockConfig) {
+    loop {
+        let vsock_backend = Arc::new(RwLock::new(
+            VhostUserVsockBackend::new(vsock_config.clone()).unwrap(),
+        ));
+
+        let listener = Listener::new(vsock_config.get_socket_path(), true).unwrap();
+
+        let mut vsock_daemon = VhostUserDaemon::new(
+            String::from("vhost-user-vsock"),
```

Comment:
I think Harsha followed what we did in QEMU where we have `vhost-user-blk`, `vhost-user-gpu`, `vhost-user-scsi`, `vhost-user-input`, etc. Honestly I would have done the same, at the most we could use `vhost-user-X-device`, though, I don't think _daemon_ adds much.

Reply:
Maybe then we should convert existing backends to use this nomenclature. I just want them to be consistent and yes daemon name doesn't matter much otherwise.

Reply:
Maybe just keep vhost-user-vsock for now ? We can discuss later in a separate pull request on how to name all of these.

Reply:
Ack